beetsplug/parentwork.py

```diff
@@ -0,0 +1,165 @@
+                if work_father['type'] == 'parts' and work_father.get(
+                    'direction') == 'backward':
+                    father_id = work_father['work']['id']
+                    partof = True
+                    work_info = musicbrainzngs.get_work_by_id(
+                        father_id,includes = ["work-rels","artist-rels"])
+    return work_info
+
+def get_info(work_info,parent_composer,parent_composer_sort,parent_work,
+    parent_work_disambig,work_ids,composer_ids):
+    """Given the parentwork info dict, this function updates the 
+    parent composer etc"""
+    composer_exists=False
+    if 'artist-relation-list' in work_info['work']:
+        for artist in work_info['work']['artist-relation-list']:
+            if artist['type'] == 'composer':
+                composer_exists=True
+                if artist['artist']['id'] not in composer_ids:
+                    composer_ids.add(artist['artist']['id'])
+                    parent_composer.append(artist['artist']['name'])
+                    parent_composer_sort.append(artist['artist']['sort-name'])
+    if not composer_exists:
+        print('no composer')
+        print('add one at')
+        print('https://musicbrainz.org/work/' + 
+            work_info['work']['id'])
+    if work_info['work']['id'] in work_ids:
+        pass
+    else:
+        parent_work.append(work_info['work']['title'])
+        work_ids.add(work_info['work']['id'])
+        if 'disambiguation' in work_info['work']:
+            parent_work_disambig.append(work_info['work']['disambiguation'])
+
+class ParentWorkPlugin(BeetsPlugin):
+
+    def __init__(self):
+        super(ParentWorkPlugin, self).__init__()
+        self.import_stages = [self.imported]
+
+    def commands(self):
+        cmd = ui.Subcommand('parentwork',
+                            help=u'fetches parent works, composers \
+                                and performers')
+        cmd.func = self.command
+        return [cmd]
+
+    def command(self, lib, opts, args):
+        self.find_work(lib.items(ui.decargs(args)))
+
+    def imported(self, session, task):
+        self.find_work(task.items)
+
+    def find_work(self, items):
+
+        
+
+        for item in items:
+            performer            = []
+            performer_sort       = []
+            work                 = []
+            work_disambig        = []
+            parent_work          = []
+            parent_work_disambig = []
+            parent_composer      = []
+            parent_composer_sort = []
+            work_ids             = set()
+            composer_ids         = set()
+            
+            item.read()
+            recording_id = item['mb_trackid']
+            performer_types = ['performer', 'instrument', 'vocal',
+                'conductor', 'performing orchestra', 'chorus master', 
+                    'concertmaster']
+            found = True
+            if item.get('parent_work')!=None:
```

Comment:
How can I read an option? I would like there to be an option '-f' to tell him to fetch the parentwork even if there is already one.